zeppelin-zengine/src/main/java/org/apache/zeppelin/notebook/repo/S3NotebookRepo.java

```diff
@@ -89,12 +92,14 @@
   private final String user;
   private final boolean useServerSideEncryption;
   private final ZeppelinConfiguration conf;
+  private boolean saveAndCommit;
 
   public S3NotebookRepo(ZeppelinConfiguration conf) throws IOException {
     this.conf = conf;
     bucketName = conf.getBucketName();
     user = conf.getUser();
     useServerSideEncryption = conf.isS3ServerSideEncryption();
+    saveAndCommit = conf.isPersistOnCommit();
```

Comment:
It doesn't make sense to ask every NotebookRepo implementation do the `saveAndCommit` logic. It is supposed to be in a base class.